Moblin/HaishinKit/Media/AudioUnit.swift

```diff
@@ -19,26 +19,120 @@ func makeChannelMap(
     return result.map { NSNumber(value: $0) }
 }
 
+protocol ReplaceAudioSampleBufferDelegate: AnyObject {
+    func didOutputReplaceSampleBuffer(cameraId: UUID, sampleBuffer: CMSampleBuffer)
+}
+
 private class ReplaceAudio {
-    var nextPresentationTimeStamp: CMTime = .zero
+    private var cameraId: UUID
+    private var latency: Double
+    private var frameRate: Double = 0.0
+    private var sampleBufferQueue: [CMSampleBuffer] = []
+    private var startTime: Double?
+    private var state: State = .initializing
+    private var initializationDuration: Double = 1
+    private var outputTimer: DispatchSourceTimer?
+    private enum State {
+        case initializing
+        case buffering
+        case outputting
+    }
+
+    private var maxQueueSize: Int {
+        return Int(latency * frameRate + 1)
```

Comment:
Is it enough to only allow a single frame more than expected to to buffered? I mean, there can be bursts of packets and frames from the RTMP publisher, which could mean we want to buffer more than 1 frame more than the target buffer size.